### PR TITLE
Protect get_terminal_size() call introduced by #1245

### PR DESCRIPTION
**Pull request by tiagocoutinho**
Fix sardana clients which don't have an interactive terminal attached to them

### CODE REVIEW

**Comment by reszelaz** [comment]
Hi Tiago,
Thanks for trying to fix the tests. There are still some problems. Meanwhile I tried to see if after reverting the #1245 the tests works and yes, I confirm that with #1269.

**Comment by reszelaz** [comment]
I have tried to run the testsuite on docker but locally on my PC instead of travis-ci and this works.
I have opened #1295 where I will try to debug this problem.

**Comment by reszelaz** [comment]
Ok, I found the problem:

```
MainThread     WARNING  2020-02-20 14:03:22,384 TaurusRootLogger: Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sardana-3.0.2a0-py3.5.egg/sardana/macroserver/macros/test/sardemoenv.py", line 192, in getElements
    elements = SarDemoEnv().getElements(elem_type)
  File "/usr/local/lib/python3.5/dist-packages/taurus/core/util/singleton.py", line 81, in __new__
    cls._the_instance.init(*p, **k)
  File "/usr/local/lib/python3.5/dist-packages/sardana-3.0.2a0-py3.5.egg/sardana/macroserver/macros/test/sardemoenv.py", line 66, in init
    self.door = Device(door_name)
  File "/usr/local/lib/python3.5/dist-packages/taurus/core/taurushelper.py", line 281, in Device
    return Factory(scheme=getSchemeFromName(device_name)).getDevice(device_name)
  File "/usr/local/lib/python3.5/dist-packages/taurus/core/tango/tangofactory.py", line 360, in getDevice
    d = dev_klass(full_dev_name, **kw)
  File "/usr/local/lib/python3.5/dist-packages/sardana-3.0.2a0-py3.5.egg/sardana/taurus/core/tango/sardana/macroserver.py", line 355, in __init__
    attr.subscribeEvent(self.logReceived, log_name)
  File "/usr/local/lib/python3.5/dist-packages/taurus/core/util/event.py", line 242, in subscribeEvent
    cb(data, self.first_event_val)
  File "/usr/local/lib/python3.5/dist-packages/sardana-3.0.2a0-py3.5.egg/sardana/taurus/core/tango/sardana/macroserver.py", line 694, in logReceived
    term_size = os.get_terminal_size()
OSError: [Errno 25] Inappropriate ioctl for device
```
You already protected this problem but it was missing to use the custom `get_terminal_size` function. I've change it in 717d60b and all the tests are green on travis.

I'm merging this. Meanwhile @rhomspuron it would be great if you could test it again in the BL22 scenario.
To protect this method Tiago added two calls: `sys.stdout.fileno()` and `os.isatty(fileno)`. I don't know if these are costly? I just tried with an `ascanct` with 3 dummy channels and could not detect any issue, but just in case it would be great if you could test it in a real scenario. In the case there are some performance issues, we could try to change the protection to simply catching `OSError`.

File: src/sardana/taurus/core/tango/sardana/macroserver.py [hunk 1]
```diff
@@ -691,7 +691,7 @@ def macroStatusReceived(self, s, t, v):
         return data
 
     def logReceived(self, log_name, output):
-        term_size = os.get_terminal_size()
+        term_size = get_terminal_size()
```
Comment (tiagocoutinho) [inline src/sardana/taurus/core/tango/sardana/macroserver.py]: Nice catch! Thanks a lot for the fix!

**Pull request merged**